src/DynamoCore/Search/NodeSearchModel.cs

```diff
@@ -231,29 +231,42 @@ internal string ProcessNodeCategory(string category, ref SearchElementGroup grou
             return category.Substring(0, index);
         }
 
-        internal IEnumerable<NodeSearchElement> Search(string search, LuceneSearchUtility luceneSearchUtility)
+        /// <summary>
+        /// Search for nodes by using a search key.
+        /// </summary>
+        /// <param name="search"></param>
+        /// <param name="luceneSearchUtility"></param>
+        /// <param name="ctk">Cancellation token to short circuit the search.</param>
+        /// <returns></returns>
+        /// <exception cref="Exception"></exception>
+        internal IEnumerable<NodeSearchElement> Search(string search, LuceneSearchUtility luceneSearchUtility, CancellationToken ctk = default)
         {
-            
+            ctk.ThrowIfCancellationRequested();
```

Comment:
where do all these get caught?

Reply:
These get caught by the Task object that wraps the parent caller
```
Task.Run(() =>
{
   return Search(query, searchCancelToken.Token);
}
```